dotnet/CoreLib/ContentStorage/AzureBlobs/AzureBlobsStorage.cs

```diff
@@ -285,6 +291,44 @@ private async Task<long> InternalWriteAsync(string directoryName, string fileNam
         return size;
     }
 
+    private async Task DeleteBlobsByPrefixAsync(string prefix, CancellationToken cancellationToken)
+    {
+        if (string.IsNullOrWhiteSpace(prefix))
+        {
+            throw new ContentStorageException("The blob prefix is empty, stopping the process to prevent data loss");
+        }
+
+        this._log.LogInformation("Deleting blobs at {0}", prefix);
+
+        AsyncPageable<BlobItem>? blobList = this._containerClient.GetBlobsAsync(prefix: prefix, cancellationToken: cancellationToken);
+        await foreach (Page<BlobItem> page in blobList.AsPages().WithCancellation(cancellationToken))
+        {
+            foreach (BlobItem blob in page.Values)
+            {
+                // Skip root and empty names
+                if (string.IsNullOrWhiteSpace(blob.Name) || blob.Name == prefix) { continue; }
+
+                // Remove the prefix, skip root and empty names
+                var fileName = blob.Name.Trim('/').Substring(prefix.Trim('/').Length).Trim('/');
+                if (string.IsNullOrWhiteSpace(fileName)) { continue; }
+
+                // Don't delete the pipeline status file
+                if (fileName == Constants.PipelineStatusFilename) { continue; }
```

Comment:
Why not delete the pipeline status file?

Reply:
deleting the file while the pipeline is in execution would break the pipeline in case of a retry. The file is updated "after" the deletion step is executed. And the corresponding message from the queue is removed "after" the step is executed. If the message remains in the queue due to some error, it would retry, not find the file, and start logging errors about a corrupted document.

Reply:
nit: should we document this somewhere?

Reply:
yes, see `CleanUpAfterCompletionAsync`